### PR TITLE
Fix documentation that led to Issue #37

### PR DESCRIPTION
**Pull request by ipundit**
[shmax](https://github.com/shmax) clarified the expected return values of Model::find() dependent on its input. See #37. This is confusing behavior for a function, but to change it would lead to breaking changes on users of this base library.

This PR does the next best thing, which is to clarify the documentation in the function header comments and also in the test cases.

Also did some cleanup of the `.gitignore` file. Usually package-lock.json would be source controlled so that you know which versions of dependencies work and when you'd have to do a `composer update` after a pull, but since composer.lock was ignored, I've also ignored package-lock.json.

### CODE REVIEW

**Comment by shmax** [comment]
> This is confusing behavior for a function, but to change it would lead to breaking changes on users of this base library.

Well, we're gearing up to do a major version release, here, so if you want to break stuff now is the time. For what it's worth there are a few other API-breaking changes that I felt were necessary for the benefit of static checking (namely, the various `validates_on` series of methods now expect an array indexed by field name, rather than a single crazy array with everything mixed-in together).

As for whether "conditions" should be a singular model or array of them, well, I have no idea.

I had to analyze this all in some depth when I was writing the PHPStan extension for `Model::find`, and it was very difficult to come up with a simple rule to govern all the cases I see in the `find` documentation.

My first thought was "ah, it's simple! If there's only one argument, then you only get one record back!". But that gets violated by this:
```php
$books = Book::find([1, 3, 8]); // array
``` 

so I thought "oh, okay, well if there's only one argument but it's an array, then we go with array". But that gets contradicted by this:
```php
$book = Book::find(['name'=>'Ubik']); // single
```

And so on. You clearly think that `["conditions"=> ...]` should be a single, but based on what rule? I'm not saying you're wrong, I'm just curious to see your gears turning.

I think convention is going to have to drive all this, but I don't really know where to go for the source of truth. This library was launched over a decade ago and was modeled after Ruby's implementation of Active Record, so if you really want to help sort all this out you might see if you can make sense of their documentation:

https://guides.rubyonrails.org/active_record_basics.html

I actually find the docs kind of scattered and hard to read, so even better might be to get Ruby working locally and maybe play around with it.

File: lib/Model.php [hunk 1]
```diff
@@ -1708,30 +1708,30 @@ public static function last(/* ... */)
      *
      * @throws RecordNotFound if no options are passed or finding by pk and no records matched
      *
-     * @return static|static[]|null
+     * @return Model|Model[]|null
```
Comment (shmax) [inline lib/Model.php:1711]: Why change this? `static` was correct--what gets returned is the late-static-bound subclass of model. So `Model` isn't wrong, exactly, but it sort of buries the lede, doesn't it?

File: lib/Model.php [hunk 2]
```diff
@@ -1765,7 +1765,7 @@ public static function find(/* $type, $options */)
 
         // find by pk
         else {
-            if (1 === count($args) && 1 == $num_args) {
+            if (1 == $num_args) {
```
Comment (shmax) [inline lib/Model.php:1768]: 👍

File: test/ActiveRecordFindTest.php [hunk 3]
```diff
@@ -19,38 +19,37 @@ public function test_find_with_no_params()
...
 
     public function test_find_returns_single_model() {
-        $author = Author::find(3);
-        $this->assertInstanceOf(Model::class, $author );
+        $author = Author::find(3, ['select' => 'author_id']);
+        $this->assertInstanceOf(Model::class, $author);
 
         $author = Author::find('3');
-        $this->assertInstanceOf(Model::class, $author );
+        $this->assertInstanceOf(Model::class, $author);
```
Comment (shmax) [inline test/ActiveRecordFindTest.php:26]: No objections to the style nit, or anything, but if they bother you, why not include the `test` folder in the `.php-cs-fixer.php` file?

File: lib/Model.php (hunk 1 above)
Comment (ipundit) [reply lib/Model.php]: Ok reverted

File: test/ActiveRecordFindTest.php (hunk 3 above)
Comment (ipundit) [reply test/ActiveRecordFindTest.php:27]: Implemented

**Comment by ipundit** [comment]
Would this be good to merge in now and then later decide on what to do for #45, and then develop a PR for that? This PR is designed to not be breaking or change the method signature in any way. Whatever is decided for #45 would likely do that.

**Review by shmax — APPROVED** [review]

**Comment by shmax** [comment]
Nice, thank you!

**Pull request merged**